library/Vanilla/Embeds/TwitchEmbed.php

```diff
@@ -0,0 +1,198 @@
+            if ($clipID['id']) {
+                $videoID = $clipID['id'];
+            }
+        }
+
+        if ($domain == "www.twitch.tv") {
+            preg_match('/(\/(?<isVideoOrCollection>\w+)\/)?(?<id>[a-zA-Z0-9]+)/i', $path, $linkType);
+            if ($linkType['isVideoOrCollection']) {
+                $this->urlType = ($linkType['isVideoOrCollection'] == 'videos') ? 'video' : 'collection';
+            } else {
+                $this->urlType = 'channel';
+            }
+            $videoID = $linkType['id'];
+        }
+        return $videoID;
+    }
+
+    /**
+     * Gets any query string attached to link.
+     *
+     * @param  string $url
+     * @return array $query
+     */
+    private function getQueryInformation($url): array {
+        $query = [];
+        $queryString = parse_url($url, PHP_URL_QUERY);
+        if ($queryString) {
+            parse_str($queryString, $query);
+        }
+        return $query;
+    }
+
+    /**
+     * Assigns a embed url based on the linktype and query string.
+     *
+     * @param string $videoID
+     * @param array $queryInfo
+     * @return string $embedUrl
+     */
+    private function getEmbedUrl($videoID, $queryInfo = null): string {
+        $embedURL = '';
+        if ($queryInfo) {
+            $t = $this->validateQueryTime($queryInfo['t']);
+            $autoplay = $this->verifyBool($queryInfo['autoplay']);
+            $muted = $this->verifyBool($queryInfo['muted']);
+        }
+
+        if ($this->urlType == 'clip' ) {
+            $embedURL = "https://clips.twitch.tv/embed?clip=".$videoID;
+        }
+
+        if ($this->urlType == 'channel') {
+            $embedURL = "https://player.twitch.tv/?channel=".$videoID;
+        }
+
+        if ($this->urlType == 'collection') {
+            $embedURL = "https://player.twitch.tv/?collection=".$videoID;
+        }
+
+        if ($this->urlType == 'video') {
+            $embedURL = "https://player.twitch.tv/?video=v".$videoID;
+            if ($autoplay) {
+                $embedURL.="&autoplay=".$autoplay;
+            }
+            if ($t) {
+                $embedURL.="&t=".$t;
+            }
+            if ($muted) {
+               $embedURL.="&muted=".$muted;
+            }
+        }
+        return $embedURL;
+    }
+
+    /**
+     * Validates the time parameter of the query string is valid.
+     *
+     * @param string $time
+     * @return string $validTime
+     */
+    private function validateQueryTime($time): string {
+        $valideTime = '';
+        if (preg_match('/\b[0-9]{1,2}h[0-9]{1,2}m[0-9]{1,2}s/i', $time, $match)) {
+            $valideTime = $match[0];
+        }
+        return $valideTime;
+    }
+
+    /**
+     * Verifies if query parameter is true or false
+     * @param string $param
+     * @return string $param
+     */
+    private function verifyBool($param): string {
+        if ((!$param == "true")||(!$param == "false")) {
```

Comment:
Spacing here. Also `==` is bad especially when comparing for truthylike strings. Use `===`.

Also in what way is this "verifying"? It doesn't return a boolean at all. It ends up returning an empty string? Why?